Vsix deploy

ryanbrandenburg opened the pull request:
### Summary of the changes

- We want our integration tests to fail if we're not running against the experimental instance we've just built. It's surprisingly difficult to tell this, but we can at least ensure that the proper Extension has been deployed (to the LocalAppData directory).
- This WON'T protect us from using an OLD experimental instance, we only know that we're using AN experimental instance. This means we need to worry about:
  - Dev-box pollution. This mostly shouldn't be a problem since most devs run Integration tests from VS (which seems to always deploy) or not at all.
  - Test machine pollution. Basically as above but for test machines.

davidwengier commented on src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/AbstractRazorEditorTest.cs:
```diff
@@ -192,6 +194,19 @@ static void WaitForFileExistsAsync(string file)
+
+            if (!version.Equals(new Version(42, 42, 42, 42)) || !assembly.Location.StartsWith(localAppData, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new NotImplementedException("Integration test not running against Experimental Extension");
```
Nit: Is this true? Or is it that it is running against the experimental instance, but the extension didn't deploy correctly?

ryanbrandenburg replied:
Hmm, I think the way I phrased it is technically accurate (in this case we don't have the Experimental Extension, just the Experimental Instance), but I can see how the phrasing is confusing. Do you have a suggestion on making it clearer?

davidwengier replied:
> we don't have the Experimental Extension, just the Experimental Instance

Thank you. I misread the last word of the message.

ryanbrandenburg commented:
So this [did achieve the goal](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6069571&view=ms.vss-test-web.build-test-results-tab&runId=38943142&resultId=100019&paneView=debug) of failing the tests when the Extension was not properly installed and used. It didn't exactly do it in the best way since every test after the first one timed out waiting for coloration instead of failing fast. I moved the check above the color check (but below the file open to ensure Extension load) and added the fix the the failure to deploy.

Apologies to those who already reviewed, I probably should have marked this as a draft PR since I was mostly trying to test failures on clean machines. Assuming the current VSMain build passes though it should be ready for re-review.

ryanbrandenburg commented on eng/pipelines/test-integration-job.yml:
```diff
@@ -29,7 +29,7 @@ steps:
   - script: eng\cibuild.cmd
       -configuration ${{ parameters.configuration }}
-      -msbuildEngine dotnet
+      -msbuildEngine vs
```
This is the fix for the failure to deploy. [Arcade does not import the targets](https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Arcade.Sdk/tools/Imports.targets#L22) required to deploy the VSIX when using the dotnet msbuildEngine.
